source/client/process_impl.cc

```diff
@@ -408,5 +409,14 @@ bool ProcessImpl::run(OutputCollector& collector) {
   return counters.find("sequencer.failed_terminations") == counters.end();
 }
 
+void ProcessImpl::setupForHRTimers() {
+  // We override the local environment to indicate to libevent that we favor precision over
+  // efficiency. Note that it is also possible to do this at setup time via libevent's api's.
+  // The upside of the approach below is that we are very loosely coupled and have a one-liner.
+  // Getting to libevent for the other approach is going to introduce more code as we would need to
+  // derive our own customized versions of certain Envoy concepts.
+  putenv(const_cast<char*>("EVENT_PRECISE_TIMER=1"));
```

Comment:
Is it safe to change this at runtime? It seems to be relying on the fact that libevent isn't caching this value or using it for its own setup. Should we at least push it to prior to wherever we initialize libevent?

Reply:
So libevent applies this [here](https://github.com/libevent/libevent/blob/13b8fc39fd4000de2c836b0acdc52ae5b4c757c7/event.c#L633).

So I think this is safe / in time for the workers.  But I do agree earlier is less risky going forward. Let me see what I can do there.

Reply:
(doing this early makes this a little more resilient to changes, and would also make this apply to the dispatcher we use above)

Reply:
Yeah, this is better. I'm still a bit sketched by using env vars here, but let's see how it goes.